automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -121,55 +121,77 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     val creatingCluster = eventually {
-      clusterCheck(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName, Seq(ClusterStatus.Creating), clusterRequest)
+//      verifyCluster(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName,
+//        Seq(ClusterStatus.Creating), clusterRequest, bucketCheck)
+      verifyCluster(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName,
+        Seq(ClusterStatus.Creating), clusterRequest)
     }(getAfterCreatePatience, implicitly[Position])
 
     if (monitor) {
       // wait for "Running", "Stopped", or error (fail fast)
       implicit val patienceConfig: PatienceConfig = clusterPatience
-      val expectedStatuses = if (clusterRequest.stopAfterCreation.getOrElse(false)) {
-        Seq(ClusterStatus.Stopped, ClusterStatus.Error)
-      } else {
-        Seq(ClusterStatus.Running, ClusterStatus.Error)
-      }
+
+      val expectedStatuses =
+        if (clusterRequest.stopAfterCreation.getOrElse(false)) {
+          Seq(ClusterStatus.Stopped, ClusterStatus.Error)
+        } else {
+          Seq(ClusterStatus.Running, ClusterStatus.Error)
+        }
+
       val runningOrErroredCluster = Try {
         eventually {
-          clusterCheck(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName, expectedStatuses, clusterRequest)
+          verifyCluster(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName,
+            expectedStatuses, clusterRequest)
         }
       }
 
       // Save the cluster init log file whether or not the cluster created successfully
       implicit val ec: ExecutionContextExecutor = ExecutionContext.global
-      saveDataprocLogFiles(creatingCluster).recover { case e =>
-        logger.error(s"Error occurred saving Dataproc log files for cluster ${creatingCluster.googleProject}/${creatingCluster.clusterName}", e)
-        None
-      }.futureValue match {
-        case Some((initLog, startupLog)) =>
+      saveDataprocLogFiles(creatingCluster).onComplete {
+        case Success(Some((initLog, startupLog))) =>
           logger.info(s"Saved Dataproc init log file for cluster ${creatingCluster.googleProject}/${creatingCluster.clusterName} to ${initLog.getAbsolutePath}")
           logger.info(s"Saved Dataproc startup log file for cluster ${creatingCluster.googleProject}/${creatingCluster.clusterName} to ${startupLog.getAbsolutePath}")
-        case None => logger.warn(s"Could not obtain Dataproc log files for cluster ${creatingCluster.googleProject}/${creatingCluster.clusterName}")
+        case Success(None) =>
+          logger.warn(s"Could not obtain Dataproc log files for cluster ${creatingCluster.googleProject}/${creatingCluster.clusterName}")
+        case Failure(e) =>
+          logger.error(s"Error occurred saving Dataproc log files for cluster ${creatingCluster.googleProject}/${creatingCluster.clusterName}", e)
+          throw e
```

Comment:
Is there a better way of accomplishing the goal of...logging the error message before failing with the exception...than re-throwing it like that?

Reply:
It seems like you got rid of the `.futureValue`. This makes the method _not_ wait for the result of the `Future`. Therefore I don't think the `throw e` would actually cause the test to fail because it's being thrown asynchronously.

In general the test code runs synchronously, so all Futures are usually attached to ta `.futureValue`. It might be better to keep the original structure, but rethrow the exception inside the `recover` block instead of returning `None`. Something like:

```
future.recover { case e =>
  log.error(...)
  throw e
}.futureValue match {
  case Some(...) => log.info("success!")
  case None => log.warn("couldn't save log files but not failing test")
}
```